### ISSUE TITLE
AccuracyMetric automatically do ReduceOp.SUM in test_epoch_end 

### ISSUE THREAD

**Issue by xiadingZ** [PyTorchLightning/pytorch-lightning#3713]
my code:
```
    def test_step(self, batch, batch_idx):
         ...
        # self.accuracy_video = Accuracy()
        vid_acc = self.accuracy_video(video_labels_hat, video_labels)
        print("test_step, ", vid_acc)

        return {'test_loss': loss, "test_pacc": part_acc, "test_vacc": vid_acc}


    def test_epoch_end(self, outputs):
        avg_loss = torch.stack([x['test_loss'] for x in outputs]).mean()
        avg_acc = torch.stack([x['test_vacc'] for x in outputs]).mean()
        print("avg_acc_1:", avg_acc)
        dist.all_reduce(avg_loss, op=dist.ReduceOp.SUM)
        dist.all_reduce(avg_acc, op=dist.ReduceOp.SUM)
        print("avg_acc_2:", avg_acc)
        avg_acc = avg_acc / self.trainer.world_size
        return {'test_loss': avg_loss, 'test_acc': avg_acc,}
```
my test_step vid_acc is 0.6~0.7, normal, but in test_epoch_end, avg_acc_1=22.3333, avg_acc_2=714.6666. Does `test_vacc`  is already synced by `dist.ReduceOp.SUM` by default?  then I only need to do `avg_acc = avg_acc / self.trainer.world_size`?

**Comment by SkafteNicki** [comment]
Yes, class metrics automatically sync between devices, with the default operation being `dist.ReduceOp.SUM`.
Depending on what version of lightning you are running, you should also be able to set the argument `reduce_op` to `'mean'` when construction the metric.

**Comment by xiadingZ** [comment]
I know it, the problem is I construct Accuracy with `reduce_op='avg'`:
```
self.accuracy_video = Accuracy(reduce_op='avg')
```
In my val step, it can normally do mean operation, value about 0.7. But in test_epoch_end, it doesn't, value about 22. Only difference is I store it in `EvalResult` in  val step, but return it directly in test step
I'm using 0.9.1 rc3
```
    def validation_step(self, batch, batch_idx):

        vid_acc = self.accuracy_video(video_labels_hat, video_labels)

        monitor = 0-vid_acc
        result = pl.EvalResult(checkpoint_on=monitor)
        result.log('val_loss', loss, sync_dist=True)
        result.log('val_vacc', vid_acc, sync_dist=True,prog_bar=True)
        result.log('monitor', monitor, sync_dist=True)

        return result


    def test_step(self, batch, batch_idx):
        vid_acc = torch.sum(video_labels == video_labels_hat).type_as(imgs) / video_labels.shape[0]

        return {'test_loss': loss, "test_pacc": part_acc, "test_vacc": vid_acc, "cm":cm}


    def test_epoch_end(self, outputs):
        avg_loss = torch.stack([x['test_loss'] for x in outputs]).mean()
        avg_acc = torch.stack([x['test_vacc'] for x in outputs]).mean()
        cm = torch.stack([x['test_vacc'] for x in outputs]).sum(dim=0)
        """
        dist.all_reduce(avg_loss, op=dist.ReduceOp.SUM)
        dist.all_reduce(avg_acc, op=dist.ReduceOp.SUM)
        dist.all_reduce(cm, op=dist.ReduceOp.SUM)
        """
        avg_loss = avg_loss / self.trainer.world_size
        avg_acc = avg_acc / self.trainer.world_size
        cm = cm / self.trainer.world_size

        print("avg_acc:", avg_acc)

```

**Comment by Evpok** [comment]
> Yes, class metrics automatically sync between devices, with the default operation being `dist.ReduceOp.SUM`.
> Depending on what version of lightning you are running, you should also be able to set the argument `reduce_op` to `'mean'` when construction the metric.

Is this mentioned in the doc? I couldn't find it there.

**Comment by SkafteNicki** [comment]
@Evpok do you refer to the auto sync or the argument?
Auto sync is mentioned a couple of times in the documentation:
https://pytorch-lightning.readthedocs.io/en/latest/metrics.html

The `reduce_op` argument is present on master, but not sure if it is mentioned in the documentation.

Please note that we are in the process of doing major changes to the metrics package before v1.0, so some of the problems mentioned here will probably be solved/made more clear after those updates.

**Comment by Evpok** [comment]
@SkafteNicki  I meant the `reduce_op` argument :-) glad to see that v1.0 should clarify it! Do you have an ETA for it?

**Comment by SkafteNicki** [comment]
@Evpok soon, cannot give exact date 👍

**Comment by williamFalcon** [comment]
fixed with new metrics interface
